README.adoc

```diff
@@ -1,22 +1,211 @@
-
-=== Retrouvez notre application en suivant le lien ci dessous
-http://collaborative-exams.eu-gb.mybluemix.net/
-
+= Première itération 
+
+*Amine* 
+
+Question
+
+* Interfaces et objets pour Question et objets associés. Une question comprend un énoncé (« phrasing »), une langue, un ensemble de réponses possibles, chacune associée à l’information indiquant si cette réponse est correcte, un identifiant (int), et un auteur. Un auteur est identifié par un id (int) et a un e-mail. Utiliser un objet Person pour les auteurs. Un énoncé et une réponse sont des textes simples. L’ensemble de réponses peut être remplacé par une information « Y/N » ou « T/F » ou « free », indiquant une question oui / non, ou vrai / faux, ou de réponse libre. Créer à cet effet une interface MultipleChoiceQuestion qui étend Question. L’auteur peut associer un identifiant personnel (string) à une question. Ceci est stocké hors de l’objet Question. Cet identifiant peut changer. (La paire (auteur, identifiant) doit être unique.) Une question est immuable. (2)
+
+
+
+
+
+
+*Remarque*
+
+Implémentation des tests unitaires associés au fur et à mesure.
+
+
+
+
+*Khaled*
+
+
+
+* Niveau de difficulté : Introduire deux nouvelle notions QuestionPhrasing et stats, deux class QuestionPhrasing (id, phrasing) et Stats(id, CountSelect, CountCorrectAnswer), par la suite la class existante Question  contiendra (id, QuestionPhrasing, Stats, … answers, …) 
+
+*Exemple : *
+Instance de QuestionPhrasing qp1 : "combien font 2+2 ?".
+Instance de Question q1 : "combien font 2+2 ?" 4 ; 6 ; 8 ; 0 ?
+Instance de Question q2 : "combien font 2+2 ?" 0 ; 4 ?
+(q1 et q2 contiennent chacune une référence vers qp1)
+Pour un examen : 
+CountSelect = nombre de candidats participants à l’examen
+CountCorrectAnswer = Un compteur pour calculer le nombre de candidats qui ont donné une bonne réponse.
+
+
+*Exemple :*
+Pour une Question 
+Exam 1 (20 participants, 10 ont donné la bonne réponse)
+ParticipateCounter = 20
+CountCorrectAnswer = 10
+Exam 1 (30 participants, 20 ont donné la bonne réponse)
+CountSelect = 20+30 = 50
+CountCorrectAnswer = 10+20 = 30
+Ainsi de suite.
+Le ratio [CountCorrectAnswer / CountSelect ] donne le niveau de difficulté de la question (0-0,33 hard; 0,33-0,66 Medium;  0,66-1 Easy)
+
+
+*Remaque*
+
+au début les question sont toutes faciles, après un certain nombre d'utilisation, le ratio aura de plus en plus de sens avec un nombre de population important.
+
+
+*Mohamed*
+
+
+* Question Xml : Représentation des questions sur un schéma XML (lecture et écriture). Chaque relation peut être représentée comme un ensemble de triplets associé à une personne
+
+* Faire les différentes conversion en Xml (QuestionText). Et ajout Tests unitaires pour GetQuestion(id).
+

```

Comment:
L’inclusion des relations va vous compliquer la tâche, trop pour une première itération, il me semble. Faites uniquement Question XML où 1) vous créez un schéma XML pour représenter juste une Question (sans ses relations), et 2) vous sérialisez et désérialisez en XML (lecture et écriture) à l’aide de JAXB. Mettez de côté (déplacez à une itération future) les relations représentées comme des triplets. Veuillez reformuler dans le texte SVP en suivant ces instructions. (Si vous avez déjà implémenté tout ça, tant mieux pour vous, on considérera que vous avez pris de l’avance sur l’itération 2, mais en tous cas, le cahier des charges doit être corrigé.)